src/runtime/moduleobject.cs

```diff
@@ -526,5 +584,25 @@ public static string[] ListAssemblies(bool verbose)
             }
             return names;
         }
+
+        /// <summary>
+        /// Note: This should *not* be called directly.
+        /// The function that get/import a CLR assembly as a python module.
+        /// This function should only be called by the import machinery as seen
+        /// in importhook.cs
+        /// </summary>
+        /// <param name="spec">A ModuleSpec Python object</param>
+        /// <returns>A new reference to the imported module, as a PyObject.</returns>
+        [ModuleFunction]
+        [ForbidPythonThreads]
```

Comment:
Why do you have to forbid python threads here?

Reply:
Because functions called from `ImportHook.Import` calls CPython internally.